tenzir/functional-test/tests/python.bats

```diff
@@ -116,32 +109,34 @@ END
 END
   check grep -v '\(Host:\|User-Agent:\)' "${data}"
   truncate -s 0 "${data}"
-}
 
+  # Teardown
+  kill -9 "$SRV_PID"
+  rm "${conn}" "${data}"
```

Comment:
If one of the commands in the test fails the rest of it won't be executed any more and the teardown here won't run, leaving the `socat` running. I believe bats copies the test contents into a temporary file and runs that with `bash -e`, so you could try to set a `trap` after creating the `coproc` instead.

Reply:
Bats recommends against setting traps, because they might interfere with the bats-internal traps.

Instead, I moved the kill back into the global teardown, and moved the directories into the temporary directory of this test so that they don't need to be removed manually.